### PR TITLE
Fix flaky zfs test

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: lib/tests.nix [hunk 1]
```diff
@@ -101,7 +101,8 @@ let
           # config for tests to make them run faster or work at all
           documentation.enable = false;
           hardware.enableAllFirmware = lib.mkForce false;
-          boot.initrd.preDeviceCommands = ''
+          # FIXME: we don't have an systemd in stage-1 equialvent for this
+          boot.initrd.preDeviceCommands = lib.mkIf (!config.boot.initrd.systemd.enable) ''
```
Comment (iFreilicht) [inline lib/tests.nix:105]: This is an unrelated drive-by, right? It doesn't seem to be related to the rest of the PR, and the test still built fine multiple times without it. What's it for? Something to do with the discussion in #651?

**Review by iFreilicht — COMMENTED** [review]
I ran it with the following script:

```
count=0; while nom build --rebuild -f tests/zfs-with-vdevs.nix; do count=$((count+1)); echo "---SUCCESS COUNT: $count ---"; done
```

After about 20 iterations, each taking 1:00-1:20 minutes, I think it's safe to say the test is no longer flaky :+1:

**Comment by Enzime** [comment]
@iFreilicht does that command fail before this PR?

**Comment by iFreilicht** [comment]
@Enzime Not fail, but the test never finishes. I've left it running for 7 minutes, and it still hangs at the reboot. After the PR, it finishes successfully consistently.

**Review by iFreilicht — APPROVED** [review]
